rpc/client_interceptors.go

```diff
@@ -45,3 +46,19 @@ func contextWithSpanMetadata(ctx context.Context) context.Context {
 	)
 	return ctx
 }
+
+// UnaryClientReauthInterceptor clears the access token stored on creds in the
+// event of an "Unauthenticated" gRPC error to force re-auth.
+func UnaryClientReauthInterceptor(creds *perRPCJWTCredentials) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{},
+		cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption,
+	) error {
+		err := invoker(ctx, method, req, reply, cc, opts...)
+		if err != nil && strings.Contains(err.Error(), "Unauthenticated") {
```

Comment:
I don't _love_ just checking for the string "Unauthenticated", and I'd rather check for the gRPC error codes 7 and 12. I tried casting to a gRPC status locally but had some trouble. Will keep looking into it, but let me know if you're familiar with how to extract the gRPC error code from a golang `error`.

Reply:
That'll do it! Thank you 🖖🏻 .

Reply:
Yeah we have some examples of this in tests... we do something like usually:

```go
       gStatus, ok := status.FromError(err)
       return ok && gStatus.Code() == codes.Unauthenticated
```

Reply:
Gotchaa thanks for the context; I did something similar but used `status.Code`:
```go
if c := status.Code(err); c == codes.Unauthenticated || c == codes.PermissionDenied &&                                                                            
       creds != nil {                                                                                                                                                  
       creds.accessToken = ""                                                                                                                                          
}       
```